runtime/jcl/common/mgmtmemory.c

```diff
@@ -702,6 +703,31 @@ Java_com_ibm_lang_management_internal_MemoryNotificationThread_processNotificati
 				return;
 			}
 
+			/* error tracepoint for the cases
+		    the value of init or max is negative but not -1; or
+		    the value of used or committed is negative; or
+		    used is greater than the value of committed; or
+		    committed is greater than the value of max if msx is not -1.
+		    */
+			for (idx = 0; idx < gcInfo->arraySize; ++idx) {
+				const char *nsGcName = (*env)->GetStringUTFChars(env, gcName, NULL);
+				char *nsPoolName = NULL;
+				if (mgmt->supportedMemoryPools > idx) {
+					nsPoolName = mgmt->memoryPools[idx].name;
+				} else {
+					nsPoolName = mgmt->nonHeapMemoryPools[idx - mgmt->supportedMemoryPools].name;
+				}
+				if ((gcInfo->preUsed[idx] > gcInfo->preCommitted[idx]) || ((-1 != gcInfo->preMax[idx]) && (gcInfo->preCommitted[idx] > gcInfo->preMax[idx])) || (0 > gcInfo->preUsed[idx]) || (0 > gcInfo->preCommitted[idx]) ||
+					(gcInfo->postUsed[idx] > gcInfo->postCommitted[idx]) ||	((-1 != gcInfo->postMax[idx]) && (gcInfo->postCommitted[idx] > gcInfo->postMax[idx])) || (0 > gcInfo->postUsed[idx]) || (0 > gcInfo->postCommitted[idx]) ||
+					((-1 != gcInfo->preMax[idx]) && (0 > gcInfo->preMax[idx])) || ((-1 != gcInfo->initialSize[idx]) && (0 > gcInfo->initialSize[idx])) || ((-1 != gcInfo->postMax[idx]) && (0 > gcInfo->postMax[idx]))) {
+					Trc_JCL_com_ibm_lang_management_internal_MemoryNotificationThread_processNotificationLoop_memoryUsageError(env, nsGcName, nsPoolName, gcInfo->initialSize[idx], gcInfo->preUsed[idx], gcInfo->preCommitted[idx], gcInfo->preMax[idx], gcInfo->postUsed[idx], gcInfo->postCommitted[idx], gcInfo->postMax[idx]);
+				}
```

Comment:
Suggest some local variables would make this more readable:
```c
    U_64 preUsed = gcInfo->preUsed[idx];
    U_64 preCommitted = gcInfo->preCommitted[idx];
    I_64 preMax = I_64 gcInfo->preMax[idx];
    // etc.
```

The condition should be split, with each `||` starting on a new line, and `) {` on a line by itself (see [1]).

The call to the tracepoint should have one argument per line (see [2]).

[1] https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#formatting-conditions
[2] https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#function-parameter-list-does-not-fit-on-one-line

Reply:
Sorry for the bad advice, `I_64 preMax = I_64 gcInfo->preMax[idx];` is not syntactically valid (remove the second `I_64`).
The condition should have the `||` at the beginning of the second and subsequent line (as documented in the coding standard referenced above).